### ISSUE TITLE
config fails: `unrecognized command line option "-Wunused-result"`

### ISSUE THREAD

**Issue by iBobik** [discoteq/flock#6]
```
[janpoboril@web470 flock-0.2.0]$ ./configure
checking for gcc... gcc
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables... 
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether we are using the GNU C compiler... yes
checking whether gcc accepts -g... yes
checking for gcc option to accept ISO C89... none needed
checking whether gcc understands -c and -o together... yes
checking how to run the C preprocessor... gcc -E
checking for grep that handles long lines and -e... /bin/grep
checking for egrep... /bin/grep -E
checking for ANSI C header files... yes
checking for sys/types.h... yes
checking for sys/stat.h... yes
checking for stdlib.h... yes
checking for string.h... yes
checking for memory.h... yes
checking for strings.h... yes
checking for inttypes.h... yes
checking for stdint.h... yes
checking for unistd.h... yes
checking minix/config.h usability... no
checking minix/config.h presence... no
checking for minix/config.h... no
checking whether it is safe to define __EXTENSIONS__... yes
checking for a BSD-compatible install... /usr/bin/install -c
checking whether build environment is sane... yes
checking for a thread-safe mkdir -p... /bin/mkdir -p
checking for gawk... gawk
checking whether make sets $(MAKE)... yes
checking for style of include used by make... GNU
checking whether make supports nested variables... yes
checking how to create a pax tar archive... gnutar
checking dependency style of gcc... gcc3
checking whether make supports nested variables... (cached) yes
checking for gawk... (cached) gawk
checking for gcc option to accept ISO C99... -std=gnu99
checking for gcc -std=gnu99 option to accept ISO Standard C... (cached) -std=gnu99
checking build system type... x86_64-unknown-linux-gnu
checking host system type... x86_64-unknown-linux-gnu
checking for an ANSI C-conforming const... yes
checking for working volatile... yes
checking whether byte ordering is bigendian... no
checking err.h usability... yes
checking err.h presence... yes
checking for err.h... yes
checking errno.h usability... yes
checking errno.h presence... yes
checking for errno.h... yes
checking fcntl.h usability... yes
checking fcntl.h presence... yes
checking for fcntl.h... yes
checking locale.h usability... yes
checking locale.h presence... yes
checking for locale.h... yes
checking signal.h usability... yes
checking signal.h presence... yes
checking for signal.h... yes
checking stdio.h usability... yes
checking stdio.h presence... yes
checking for stdio.h... yes
checking for stdlib.h... (cached) yes
checking for string.h... (cached) yes
checking sys/time.h usability... yes
checking sys/time.h presence... yes
checking for sys/time.h... yes
checking sysexits.h usability... yes
checking sysexits.h presence... yes
checking for sysexits.h... yes
checking for unistd.h... (cached) yes
checking for err... yes
checking for errx... yes
checking for warn... yes
checking for warnx... yes
checking for flock... yes
checking that generated files are newer than configure... done
configure: creating ./config.status
config.status: creating Makefile
config.status: creating config.h
config.status: executing depfiles commands
[janpoboril@web470 flock-0.2.0]$ make
make  all-am
make[1]: Entering directory `/home/janpoboril/flock-0.2.0'
  CC       src/flock.o
cc1: error: unrecognized command line option "-Wunused-result"
make[1]: *** [src/flock.o] Error 1
make[1]: Leaving directory `/home/janpoboril/flock-0.2.0'
make: *** [all] Error 2
```

**Comment by josephholsten** [comment]
ooh, that's no good. Let me see if I can reproduce.

**Comment by josephholsten** [comment]
ah, well I'm getting a totally different error on debian 8.2.2, but let's see if me fixing that helps you as well.

**Comment by iBobik** [comment]
Just for record: CentOS release 6.7 (Final)



Jan Pobořil

2015-12-18 18:50 GMT+01:00 Joseph Anthony Pasquale Holsten <
notifications@github.com>:

**Comment by josephholsten** [comment]
ok, I'll check that next! I've also just pushed v0.2.1 which should fix the build on some linux systems. I didn't touch anything related to `-Wunused-result` though, so I'm not sure it'll help you yet.

**Comment by josephholsten** [comment]
ok, I had a typo in `v0.2.1`, but `v0.2.2` seems to work on CentOS 7. Grabbing a CentOS 6.7 image now.

**Comment by josephholsten** [comment]
ok, got the error reproduced now. looks like I was a little over-zealous with the automake cflags. Patch incoming!

**Comment by josephholsten** [comment]
ok, this should be fixed in `v0.2.3`. The readme has the updated artifact link, I was able to just copy its commands into my terminal:

```
FLOCK_VERSION=0.2.3
wget https://github.com/discoteq/flock/releases/download/v${FLOCK_VERSION}/flock-${FLOCK_VERSION}.tar.xz
xz -dc flock-${FLOCK_VERSION}.tar.xz | tar -x
cd flock-${FLOCK_VERSION}
./configure
make
make install
```

Re-open if you still see any issues!
